packages/amplify-category-storage/src/provider-utils/awscloudformation/service-walkthroughs/s3-user-input-state.ts

```diff
@@ -0,0 +1,163 @@
+import { S3AccessType, S3PermissionType, S3UserInputs } from '../service-walkthrough-types/s3-user-input-types';
+import { AmplifyCategories, AmplifySupportedService } from 'amplify-cli-core';
+import { JSONUtilities, pathManager } from 'amplify-cli-core';
+import { CLIInputSchemaValidator } from 'amplify-cli-core';
+
+import * as fs from 'fs-extra';
+import * as path from 'path';
+
+type ResourcRefType =  {
+    Ref: string
+}
+
+export enum S3CFNPermissionType {
+    PUT_OBJECT = "s3:PutObject",
+    GET_OBJECT = "s3:GetObject",
+    DELETE_OBJECT = "s3:DeleteObject",
+    LIST_BUCKET = "s3:ListBucket"
+}
+
+export type S3CFNDependsOn = {
+     category : string,
+     resourceName : string,
+     attributes : string[]
+}
+
+export type S3CLIWalkthroughParams = {
```

Comment:
This is also missing user pool groups permissions types.

Reply:
Why do we need this type? Can we directly have this flow:

`Prompts customers for questions -> directly convert to cli-inputs.json type  (which is defined below) -> convert to props exposed to overrides file -> use these props to generate CFN using CDK.`

Currently we are maintaining two layers of state within the code which I think we should be avoiding.